src/main/java/com/coupang/marketplace/service/UserService.java

```diff
@@ -32,4 +38,24 @@ public void join(SignUpRequestDto dto){
     public boolean checkIsUserExist (String email) {
         return userRepository.findByEmail(email).isPresent();
     }
+
+    public void login(SignInRequestDto dto){
+        if (!checkIsUserExist(dto.getEmail())) {
+            throw new IllegalArgumentException("존재하지 않는 이메일입니다.");
+        }
+        else {
+            Optional<User> user = userRepository.findByEmail(dto.getEmail());
+            String userSalt = user.get().getSalt();
+
+            String password = dto.getPassword();
+            String encryptedPassword = sha256Encryptor.run(password, userSalt);
+
+            if(encryptedPassword.equals(user.get().getPassword())){
+                httpSession.setAttribute(SessionKey.LOGIN_USER_ID, user.get().getId());
+            }
+            else{
+                throw new IllegalArgumentException("패스워드가 틀렸습니다.");
+            }
```

Comment:
```suggestion
            if(!encryptedPassword.equals(user.get().getPassword())){
            	throw new IllegalArgumentException("패스워드가 틀렸습니다.");
            }
            httpSession.setAttribute(SessionKey.LOGIN_USER_ID, user.get().getId());
```
이렇게 쓴다면 더 간결해지지 않을까요? 😀 제안입니다!

Reply:
아 그렇겠네요! 수정하도록 하겠습니다😊